client/src/components/checkbox-radio-field/index.tsx

```diff
@@ -9,17 +9,21 @@ import { ICheckboxProps } from '../checkbox-radio/types';
 
 export const CheckboxRadioField = ({
   name,
+  type,
   label,
   ...props
 }: ICheckboxProps) => {
   return (
     <Field name={name}>
-      {({ meta, form, field: { ...fieldProps } }: FieldProps) => {
+      {({ meta, form, field: { value, ...fieldProps } }: FieldProps) => {
         const error = Error.controlError(meta, name, label);
+        const checked = type === 'radio' ? value === label : value;
 
         const change = async (e: ChangeEvent<HTMLInputElement>) => {
           if (name) {
-            await form.setFieldValue(name, e.target.checked);
+            const value = type === 'radio' ? e.target.value : e.target.checked;
```

Comment:
The name of the variable doesn't explain what is the content or purpose of the value in it.